### PR TITLE
feat: add <Button /> component

### PR DESCRIPTION
**Pull request by satya164**
**TODO:**
- [ ] Ability to customize text color
- [ ] Disabled button
- [ ] Button with icon
- [ ] Button with loading indicator

image

image

### CODE REVIEW

**Comment by ahmedlhanafy** [comment]
@satya164 looking good i will test once I'm on my computer and merge

File: src/components/Typography/Text.js [hunk 1]
```diff
@@ -19,6 +19,12 @@ class Text extends Component<void, Props, void> {
     style: NativeText.propTypes.style,
   };
 
+  _root: any;
+
+  setNativeProps(...args) {
```
Comment (ahmedlhanafy) [inline src/components/Typography/Text.js]: What is the purpose of this? @satya164

Comment (satya164) [inline src/components/Typography/Text.js]: @ahmedlhanafy `TouchableOpacity` and `TouchableHighlight` need it to have this as a direct child.

Comment (ahmedlhanafy) [inline src/components/Typography/Text.js]: @satya164 won't they be available by extending `Component` only?

Comment (ahmedlhanafy) [inline src/components/Typography/Text.js]: Cool

**Pull request merged**